test/extended/images/db_helpers.go

```diff
@@ -0,0 +1,120 @@
+package images
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+// MySQL is a MySQL helper for executing commands
+type MySQL struct {
+	oc        *exutil.CLI
+	PodName   string
+	Container string
+	Env       map[string]string
+}
+
+// NewMysql queries OpenShift for a pod with given name, saving environment
+// variables like username and password for easier use.
+func NewMysql(oc *exutil.CLI, podName, masterPodName string) (*MySQL, error) {
+	m := &MySQL{
+		oc:      oc,
+		PodName: podName,
+	}
+
+	if pod, err := oc.KubeREST().Pods(oc.Namespace()).Get(podName); err != nil {
+		return nil, err
+	} else {
+		m.Container = pod.Spec.Containers[0].Name
+	}
+
+	if masterPodName != "" {
+		// Some values are replicated to the master, we'll need to get them
+		if err := m.GetEnv(masterPodName); err != nil {
+			return nil, err
+		}
+	}
+	if err := m.GetEnv(podName); err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
+// GetEnv will get and save environment variables from a pod
+func (m *MySQL) GetEnv(podName string) error {
+	pod, err := m.oc.KubeREST().Pods(m.oc.Namespace()).Get(podName)
+	if err != nil {
+		return err
+	}
+	if m.Env == nil {
+		m.Env = make(map[string]string)
+	}
+	for _, container := range pod.Spec.Containers {
+		for _, env := range container.Env {
+			m.Env[env.Name] = env.Value
+		}
+	}
+	return nil
+}
+
+// WaitForMySQL pings the MySQL server
+func (m *MySQL) WaitForMySQL() (bool, error) {
```

Comment:
move this away from MySQL struct, make it a standalone helper. podname, containername and oc as arguments.